### PR TITLE
Expose new CLI option in CheckFingerprint 

### PR DESCRIPTION
**Pull request by SHuang-Broad**
allowing user to set a minimal base quality to be used for checking.

### Description

Some type of long read data have base qualities all set to 0 from the sequencer.
We are proposing to allow `CheckFingerprint` to use a custom minimal base quality&mdash;an option available to `FingerprintChecker` but not exposed through CLI.
We've tested `CheckFingerprint` on these types of data with this update, and it works mostly as expected (there's no guarantee, and honestly, base qual all == 0 is the root cause, which we have no control over).

----

#### Content
- [x] Added or modified tests to cover changes and any new functionality
- [ ] (NA) Edited the README / documentation (if applicable)
- [ ] All tests passing on Travis

#### Review
- [ ] Final thumbs-up from reviewer
- [ ] Rebase, squash and reword as applicable

### CODE REVIEW

File: src/test/java/picard/fingerprint/CheckFingerprintTest.java [hunk 1]
```diff
@@ -272,21 +288,45 @@ void testCheckFingerprintNoRg(final File file, final File genotypes, final int e
...
+
+        return args;
+    }
+
+    private File tester(boolean ignoreRG, final File file, final File genotypes, final int expectedRetVal) throws IOException {
+        final List<String> args = buildArgs(ignoreRG, file, genotypes);
+        final File outputSummary = File.createTempFile("fingerprint", "summary_metrics");
+        outputSummary.deleteOnExit();
+        final File outputDetail = File.createTempFile("fingerprint", "detail_metrics");
+        outputSummary.deleteOnExit();
```
Comment (gbggrant) [inline src/test/java/picard/fingerprint/CheckFingerprintTest.java:309]:
```suggestion
        outputDetail.deleteOnExit();
```

File: src/test/java/picard/fingerprint/CheckFingerprintTest.java [hunk 2]
```diff
@@ -272,21 +288,45 @@ void testCheckFingerprintNoRg(final File file, final File genotypes, final int e
...
+
+        return outputSummary;
+    }
+
+    private File testWithMinBaseQ(boolean ignoreRG, final File file, final File genotypes, final int expectedRetVal, final int minBaseQ) throws IOException {
+        final List<String> args = buildArgs(ignoreRG, file, genotypes);
+        final File outputSummary = File.createTempFile("fingerprint", "summary_metrics");
+        outputSummary.deleteOnExit();
+        final File outputDetail = File.createTempFile("fingerprint", "detail_metrics");
+        outputSummary.deleteOnExit();
```
Comment (gbggrant) [inline src/test/java/picard/fingerprint/CheckFingerprintTest.java:326]:
```suggestion
        outputDetail.deleteOnExit();
```

**Review by gbggrant — APPROVED** [review]
Looks good - two minor errors in the test.

**Comment by gbggrant** [comment]
Looks good @kachulis to you?

**Comment by SHuang-Broad** [comment]
Thanks! @gbggrant 
Sorry I wasn't up to speed on this.

File: src/test/java/picard/fingerprint/CheckFingerprintTest.java [hunk 3]
```diff
@@ -272,21 +288,45 @@ void testCheckFingerprintNoRg(final File file, final File genotypes, final int e
...
+
+        Assert.assertEquals(runPicardCommandLine(args), expectedRetVal);
+
+        Assert.assertTrue(outputSummary.exists(), "Expected output file " + outputSummary.getAbsolutePath() + " to exist.");
+        Assert.assertTrue(outputDetail.exists(), "Expected output file " + outputDetail.getAbsolutePath() + " to exist.");
+
+        return outputSummary;
+    }
+
+    private File testWithMinBaseQ(boolean ignoreRG, final File file, final File genotypes, final int expectedRetVal, final int minBaseQ) throws IOException {
```
Comment (kachulis) [inline src/test/java/picard/fingerprint/CheckFingerprintTest.java:321]: since this is basically identical to `tester` what about adding an array input of `additionalArgs` to tests that gets appended to the arguments, and just using that to reduce the duplicate code

File: src/test/java/picard/fingerprint/CheckFingerprintTest.java [hunk 4]
```diff
@@ -272,21 +288,45 @@ void testCheckFingerprintNoRg(final File file, final File genotypes, final int e
...
+        return args;
+    }
+
+    private File tester(boolean ignoreRG, final File file, final File genotypes, final int expectedRetVal) throws IOException {
+        final List<String> args = buildArgs(ignoreRG, file, genotypes);
+        final File outputSummary = File.createTempFile("fingerprint", "summary_metrics");
+        outputSummary.deleteOnExit();
+        final File outputDetail = File.createTempFile("fingerprint", "detail_metrics");
+        outputDetail.deleteOnExit();
+        args.add("SUMMARY_OUTPUT=" + outputSummary.getAbsolutePath());
```
Comment (kachulis) [inline src/test/java/picard/fingerprint/CheckFingerprintTest.java:310]: cleaner to bring these inside buildArgs, no?

**Review by kachulis — COMMENTED** [review]
looks good, just some minor comments.  Also wondering if it would be good to expose the same option in crosscheckfingerprints, since I imagine you'd hit the same issue if you tried to use it with this data.

File: src/main/java/picard/fingerprint/CheckFingerprint.java [hunk 5]
```diff
@@ -364,6 +369,12 @@ protected int doWork() {
             if (fileContainsReads && OBSERVED_SAMPLE_ALIAS != null) {
                 return new String[]{"The parameter OBSERVED_SAMPLE_ALIAS can only be used with a VCF input."};
             }
+
+            if (fileContainsReads) {
+                if (MIN_BASE_QUAL < 0 || (MIN_BASE_QUAL > 60)) {
```
[the remark this answers is not in the document]
Comment (SHuang-Broad) [reply src/main/java/picard/fingerprint/CheckFingerprint.java:374]: That's a fair question. Ideally, I'd raise a warning, as most likely >60 is unintended.

**Comment by gbggrant** [comment]
@SHuang-Broad - can you address the suggestions that @kachulis has made?

**Comment by kachulis** [comment]
@SHuang-Broad this looks good once you remove the MIN_BASE_QUALITY>60 requirement.

**Comment by SHuang-Broad** [comment]
closing as the type of data which needs this feature is phased out.
Thanks for the review comments!
